src/fhir.js

```diff
@@ -384,21 +384,35 @@ class Patient extends FHIRObject {
     }
   }
 
-  findRecords(profile) {
-    const classInfo = this._modelInfo.findClass(profile);
-    if (classInfo == null) {
-      console.error(`Failed to find type info for ${profile}`);
-      return [];
+  findRecords(profile, retrievedDetails) {
+    let classInfo;
+    if (retrievedDetails) {
+      classInfo = this._modelInfo.findClass(retrievedDetails.dataType);
+      const resourceType = classInfo.name.replace(/^FHIR\./, '');
+      const records = this._bundle.entry
+        .filter(e => {
+          return e.resource && e.resource.resourceType == resourceType;
+        })
+        .map(e => {
+          return new FHIRObject(e.resource, classInfo, this._modelInfo);
+        });
+      return records;
```

Comment:
We repeat this entire block of code in both the if and the else, can we rework this such that the if/else only affects the assignment of the `classInfo` variable? The rest should be the same, regardless of whether or not retreiveDetails was used